### DISCUSSION TITLE
Hide window titlebar for desktop platforms

### DISCUSSION THREAD

**Discussion by ItsCubeTime** [beeware/toga#1792] [Q&A]
Is it correct that you cannot hide the window titlebar for desktop platforms currently?

![image](https://user-images.githubusercontent.com/20190653/220474660-924276df-cafb-4d4a-8695-4171332f7d59.png)
Im using Toga as a PWA window host via the WebView widget, would like to be able to use the WebView widget to design my own titlebar:
![image](https://user-images.githubusercontent.com/20190653/220475372-2f03701d-9da5-4be3-a6a4-ceef3f264c64.png)

**Comment by freakboy3742** [comment]
(Converted to a discussion question)

You're correct that there isn't currently a way to hide the menubar - all apps have a menu bar. 

#733 is tracking a feature request to make this optional.

**Reply by ItsCubeTime** [reply]
I see, but isnt the menubar the menu below the titlebar of the window? Just wanting to make sure its *really* the same request. In my case I want both to go

It also seems the workaround they mention in the thread isnt working anymore?

**Reply by freakboy3742** [reply]
The position of the menubar is platform dependent. On macOS, it's not in the app window, because that's not where macOS puts menus. On Windows and Linux, it's the menu under the titlebar in the main window.

As for the workaround not working - that's the nature of workarounds :-)

**Reply by mhsmith** [reply]
To answer the original question: Toga doesn't have a way of hiding *title* bars either, and since this is a fairly unusual feature, we probably won't be adding it soon.

**Reply by freakboy3742** [reply]
Oh! I completely misunderstood - if you indeed meant removing the entire title bar of the window, then (a) there's no way to do that, and (b) we're unlikely to ever add such a feature, as it's fundamentally in opposition to the goal of Toga apps adhering to local operating system appearance.

**Comment by dhbmarcos** [comment]
For hide menubar in Linux (Gnome 3) I have used:

~~~python
import toga

class MainWindow(toga.App):

    def startup(self):
        self.commands.clear()
        self.main_window = toga.MainWindow(title=self.formal_name)
        self.main_window.show()

def main():
    return MainWindow()

~~~

The line 
~~~ python
self.commands.clear()
~~~
remove the menu bar from App.

In the Gnome Style Guide (Linux) and Material Design (Android) have a option to remove menu bar.

**Reply by freakboy3742** [reply]
@dhbmarcos You may want to check out PR #2244 - that's converging on an official API for removing the menubar, and a couple of of related features.
